docs/source/install/puppet.rst

```diff
@@ -0,0 +1,311 @@
+Puppet Module
+=============
+
+If you're ready to take complete control of your |st2| instances, then ``puppet-st2``
+is for you! It offers repeatable, configurable, and idempotent production-friendly
+|st2| installations.
+
+The source code for all the Puppet module is available as a GitHub repo:
+`StackStorm/puppet-st2 <https://github.com/stackstorm/puppet-st2/>`_
+
+.. contents:: Contents
+   :local:
+
+---------------------------
+
+Supported Platforms
+-------------------
+
+The Puppet module supports the same platforms as manual installation, i.e.:
+
+* Ubuntu Trusty (14.04)
+* Ubuntu Xenial (16.04)
+* RHEL 6/CentOS 6
+* RHEL 7/CentOS 7
+
+The same system size :doc:`requirements </install/system_requirements>` also apply.
+
+Quick Start
+-----------
+
+The first step is installing Puppet, for this please consule the
+`Puppet Installation documentation <https://puppet.com/docs/puppet/latest/install_linux.html>`_
+
+To get started with a single node deployment, and default configuration settings, run these
+commands (expected to run as `root`):
+
+.. note::
+
+  Puppet versions <= 3.x are no longer supported. Please utilize Puppet >= 4.
+
+First, create a file called ``Puppetfile`` with the following content:
+
+.. code-block:: ruby
+
+  forge 'https://forgeapi.puppetlabs.com'
+  mod 'stackstorm/st2'
+
+Next, utilize ``librarian-puppet`` to download and install the ``puppet-st2`` module
```

Comment:
Wondering if we could minimize this part, including manual instructions with librarian, which looks a bit hacky.

Look at https://github.com/StackStorm/st2docs/blob/f5658ffc3e8ee4846d9faec77dbed182c538f663/docs/source/install/puppet_chef_salt_ansible.rst#manual-installation if we could take examples of the previous Puppet installation instructions.

For example, is 
```
puppet module install stackstorm-st2
```
still a possible thing?

Reply:
Since I'm not familiar with Puppet, curious if `Quick Start` could be really quick :)